kaldb/src/main/java/com/slack/kaldb/preprocessor/PreprocessorService.java

```diff
@@ -337,6 +337,12 @@ protected static Properties makeKafkaStreamsProps(
     props.put(
         StreamsConfig.PROCESSING_GUARANTEE_CONFIG, kafkaStreamConfig.getProcessingGuarantee());
 
+    // don't override any property we already set
+    for (Map.Entry<String, String> additionalProp :
+        kafkaStreamConfig.getAdditionalPropsMap().entrySet()) {
+      maybeOverride(props, additionalProp.getKey(), additionalProp.getValue(), false);
```

Comment:
We don't necessarily need the `maybeOverride` method - we can just set the user values first and then set our values after. If you still wanted a log message you can just rely on the fact that putting a value returns the previous value if it existed, else null.

Reply:
I didn't want the user to override thing like acks, replication etc. So I thought this approach would be nicer.

Eventually we can have a list of params that are overridable and for those pass true